lib/active_merchant/billing/gateways/rapyd.rb

```diff
@@ -58,26 +58,37 @@ def authorize(money, payment, options = {})
 
       def capture(money, authorization, options = {})
         post = {}
-        commit(:post, "payments/#{authorization}/capture", post)
+        commit(:post, "payments/#{add_reference(authorization)}/capture", post)
       end
 
       def refund(money, authorization, options = {})
         post = {}
-        post[:payment] = authorization
+        post[:payment] = add_reference(authorization)
         add_invoice(post, money, options)
         add_metadata(post, options)
         commit(:post, 'refunds', post)
       end
 
       def void(authorization, options = {})
         post = {}
-        commit(:delete, "payments/#{authorization}", post)
+        commit(:delete, "payments/#{add_reference(authorization)}", post)
       end
 
       def verify(credit_card, options = {})
         authorize(0, credit_card, options)
       end
 
+      def store(payment, options = {})
+        post = {}
+        add_payment(post, payment, options)
+        add_customer_object(post, payment, options)
+        commit(:post, 'customers', post)
+      end
+
+      def unstore(customer)
```

Comment:
Is this `customer` value being passed in by the merchant as a gateway-specific field?

Reply:
Thanks for this call out. `customer` should just be the reference to the `store` transaction (which in Rapyd's case, should be the customer token). I modified this call to pass the `customer` to `add_reference` in order to properly extract that value.